components/TableData/TableData.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+'use client';
+
+import { JSX, useState } from 'react';
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+  SortingState,
+  getSortedRowModel,
+} from '@tanstack/react-table';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '../Table/Table';
+import { ITableData } from './TableData.interface';
+
+/**
+ * Alert Notification pop ups.
+ * @param props - The page props.
+ * @param props.columns - Each column of the data table.
+ * @param props.data - Data that goes inside the table.
+ * @returns {JSX.Element} The rendered weekly picks page.
+ */
+const TableData = <TData, TValue>({
+  columns,
+  data,
+}: ITableData<TData, TValue>): JSX.Element => {
+  const [sorting, setSorting] = useState<SortingState>([]);
+
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+    onSortingChange: setSorting,
+    getSortedRowModel: getSortedRowModel(),
+    state: {
+      sorting,
+    },
+  });
+
+  return (
+    <div className="rounded-md border">
+      <Table>
+        <TableHeader>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <TableRow key={headerGroup.id}>
+              {headerGroup.headers.map((header) => {
+                return (
+                  <TableHead key={header.id}>
+                    {header.isPlaceholder
+                      ? null
+                      : flexRender(
+                        header.column.columnDef.header,
+                        header.getContext(),
+                      )}
+                  </TableHead>
+                );
+              })}
+            </TableRow>
+          ))}
+        </TableHeader>
+        <TableBody>
+          {table.getRowModel().rows?.length ? (
```

Comment:
Create a const for `table.getRowModel().rows` so you don't need to call it twice.